controller/apps.go

```diff
@@ -0,0 +1,443 @@
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/fabric8-services/fabric8-wit/auth/authservice"
+
+	"github.com/fabric8-services/fabric8-wit/app"
+	"github.com/fabric8-services/fabric8-wit/auth"
+	"github.com/fabric8-services/fabric8-wit/configuration"
+	witerrors "github.com/fabric8-services/fabric8-wit/errors"
+	"github.com/fabric8-services/fabric8-wit/kubernetes"
+	"github.com/fabric8-services/fabric8-wit/log"
+	"github.com/goadesign/goa"
+	uuid "github.com/satori/go.uuid"
+)
+
+// AppsController implements the apps resource.
+type AppsController struct {
+	*goa.Controller
+	Config *configuration.Registry
+}
+
+// NewAppsController creates a apps controller.
+func NewAppsController(service *goa.Service, config *configuration.Registry) *AppsController {
+	return &AppsController{
+		Controller: service.NewController("AppsController"),
+		Config:     config,
+	}
+}
+
+func tostring(item interface{}) string {
+	bytes, _ := json.MarshalIndent(item, "", "  ")
+	return string(bytes)
+}
+
+func (c *AppsController) getAndCheckOSIOClient(ctx context.Context) *OSIOClient {
+
+	// TODO - grab the WIT URL from the incoming request if possible
+	witURL := "localhost"
+
+	// TODO - remove this debug hook before production
+	if os.Getenv("OSIO_WIT_URL") != "" {
+		witURL = os.Getenv("OSIO_WIT_URL")
+	}
+
+	oc := NewOSIOClient(ctx, witURL)
+
+	return oc
+}
+
+func (c *AppsController) getSpaceNameFromSpaceID(ctx context.Context, spaceID uuid.UUID) (*string, error) {
+	// TODO - add a cache in AppsController
+	// use WIT API to convert Space UUID to Space name
+	oc := c.getAndCheckOSIOClient(ctx)
+
+	osioSpace, err := oc.GetSpaceByID(ctx, spaceID)
+	if err != nil {
+		return nil, err
+	}
+	return osioSpace.Attributes.Name, nil
+}
+
+func (c *AppsController) getNamespaceName(ctx context.Context) (*string, error) {
+
+	osioclient := c.getAndCheckOSIOClient(ctx)
+	kubeSpaceAttr, err := osioclient.GetNamespaceByType(ctx, nil, "user")
+	if err != nil {
+		return nil, err
+	}
+	if kubeSpaceAttr == nil || kubeSpaceAttr.Name == nil {
+		return nil, witerrors.NewNotFoundError("namespace", "user")
+	}
+
+	return kubeSpaceAttr.Name, nil
+}
+
+func (c *AppsController) getUser(authClient authservice.Client, ctx context.Context) (*authservice.User, error) {
+	// get the user definition (for cluster URL)
+	resp, err := authClient.ShowUser(ctx, authservice.ShowUserPath(), nil, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+
+	status := resp.StatusCode
+	if status < 200 || status > 300 {
+		return nil, errors.New("Failed to GET user due to status code " + string(status))
+	}
+
+	respBody, err := ioutil.ReadAll(resp.Body)
+
+	var respType authservice.User
+	err = json.Unmarshal(respBody, &respType)
+	if err != nil {
+		return nil, err
+	}
+	return &respType, nil
+}
+
+func (c *AppsController) getToken(authClient authservice.Client, ctx context.Context, forService string) (*authservice.TokenData, error) {
```

Comment:
this could be a function, I believe (ie, no need for the method receiver, it doesn't seem to be used) - but that's just a remark